src/textarea/tests/functional/Textarea.ts

```diff
@@ -0,0 +1,155 @@
+	},
+	'label should be as defined'() {
+		return getPage(this.remote)
+			.findByCssSelector(`.t1.${css.root}`)
+				.getVisibleText()
+				.then(text => {
+					assert.strictEqual(text, 'Type Something');
+				})
+			.end();
+	},
+	'should gain focus when clicking on the label'() {
+		let elementClassName: string;
+		return getPage(this.remote)
+			.findByCssSelector(`.t1.${css.root}`)
+				.findByCssSelector(`.${css.input}`)
+					.getProperty('className')
+					.then((className: string) => {
+						elementClassName = className;
+					})
+				.end()
+				.click()
+				.sleep(1000)
+				.getActiveElement()
+				.then(element => {
+					return element
+						.getProperty('className')
+						.then((className: string) => {
+							assert.strictEqual(className, elementClassName);
+						});
+				})
+			.end();
+	},
+	'should allow input to be typed'() {
+		const testInput = 'test text';
+		return getPage(this.remote)
+			.findByCssSelector(`.t1.${css.root}`)
+				.click()
+				.findByCssSelector(`.${css.input}`)
+					.type(testInput)
+					.getProperty('value')
+					.then((value: string) => {
+						assert.strictEqual(value, testInput);
+					})
+				.end()
+			.end();
+	},
+	'disabled should not allow input to be typed'() {
+		const initValue = 'Initial value';
+		return getPage(this.remote)
+			.findByCssSelector(`.t2.${css.root} .${css.input}`)
+				.click()
+				.then(null, () => {})
+				.type('text')
+				.then(null, () => {})
+				.getProperty('value')
+				.then((value: string) => {
+					assert.strictEqual(value, initValue);
+				})
+			.end();
+	},
+	'validated should update style based on validity'() {
+		const validText = 'exists';
+		const backspaces = [];
+		for (let i = 0; i < validText.length; i++) {
+			backspaces.push(keys.BACKSPACE);
+		}
+
+		return getPage(this.remote)
+			.findByCssSelector(`.t3.${css.root}`)
+				.getProperty('className')
+				.then((className: string) => {
+					assert.notInclude(className, css.invalid);
+					assert.notInclude(className, css.valid);
+				})
+				.findByCssSelector(`.${css.input}`)
+					.click()
+					.type(validText)
+				.end()
+			.end()
+			// focus another input
+			.findByCssSelector(`.t1.${css.root} .${css.input}`)
+				.click()
+			.end()
+			.sleep(500)
+			// enter invalid value
+			.findByCssSelector(`.t3.${css.root}`)
+				.getProperty('className')
+				.then((className: string) => {
+					assert.notInclude(className, css.invalid);
+					assert.include(className, css.valid);
+				})
+				.findByCssSelector(`.${css.input}`)
+					.click()
+					.type(backspaces)
+				.end()
+			.end()
+			// focus another input
+			.findByCssSelector(`.t1.${css.root} .${css.input}`)
+				.click()
+			.end()
+			.sleep(500)
+			.findByCssSelector(`.t3.${css.root}`)
+				.getProperty('className')
+				.then((className: string) => {
+					assert.notInclude(className, css.valid);
+					assert.include(className, css.invalid);
+				})
+			.end();
+	},
+	'hidden label should not be displayed'() {
```

Comment:
I'm not sure we need to add an example and test for this, since it's part of label functionality rather than textarea functionality (and I'm not a fan of hidden labels, so not big on modeling them in all the widget examples)